test/e2e/kubectl.go

```diff
@@ -188,6 +195,114 @@ var _ = Describe("Kubectl client", func() {
+
+			// Set up the form file
+			fileWriter, err := postBodyWriter.CreateFormFile("file", "kubectl")
+			if err != nil {
+				Failf("Unable to to write kubectl at %s to buffer. Error: %s", kubectlBinPath, err)
+			}
+
+			// Open up the kubectl binary for reading
+			fh, err := os.Open(kubectlBinPath)
+			if err != nil {
+				Failf("Unable to to access kubectl at %s. Error: %s", kubectlBinPath, err)
+			}
+
+			// Copy kubectl binary into the file writer
+			_, err = io.Copy(fileWriter, fh)
+			if err != nil {
+				Failf("Unable to to copy kubectl at %s into the file writer. Error: %s", kubectlBinPath, err)
+			}
+			// Nothing more should be written to this instance of the postBodyWriter
+			postBodyWriter.Close()
+
+			resp, err := http.Post(fmt.Sprintf("%s/upload", netexecAddr), postBodyWriter.FormDataContentType(), postBodyBuffer)
+
+			uploadBinaryName, err := ioutil.ReadAll(resp.Body)
+			resp.Body.Close()
+			if err != nil {
+				Failf("Unable to read the name of the uploaded binary: %s", err)
+			}
+
+			if resp.StatusCode != http.StatusCreated {
+				Failf("Unable to upload kubectl binary to remote exec server. Expected status: 201, got %d", resp.StatusCode)
+			}
+			if err != nil {
+				Failf("Unable to upload kubectl binary to the remote exec server due to error: %s", err)
+			}
+			for _, proxyVar := range []string{"http_proxy", "HTTP_PROXY", "https_proxy", "HTTPS_PROXY"} {
+				shellCommand := fmt.Sprintf("%s=%s %s --server=%s --namespace=%s exec nginx echo running in container", proxyVar, proxyAddr, uploadBinaryName, apiserverAddr, ns)
+				// Execute kubectl on remote exec server.
+				resp, err = http.PostForm(fmt.Sprintf("%s/shell", netexecAddr), url.Values{"shellCommand": {shellCommand}})
+				if err != nil {
+					Failf("Unable to execute kubectl binary on the remote exec server due to error: %s", err)
+				}
+				if resp.StatusCode != http.StatusOK {
+					errOut, _ := ioutil.ReadAll(resp.Body)
+					Failf("Unable to execute kubectl binary on remote exec server. Expected status: %d, got %d. Info: %s", http.StatusCreated, resp.StatusCode, errOut)
+				}
+
+				proxyExecOutput, err := ioutil.ReadAll(resp.Body)
+				if err != nil {
+					Failf("Unable to read the result of the kubectl execution on the remote exec server due to error: %s", err)
+				}
+				resp.Body.Close()
+				expectedExecOutput := "running in container\n"
+				// Verify we got the normal output captured by the exec server
+				if string(proxyExecOutput) != expectedExecOutput {
+					Failf("Unexpected kubectl exec output. Wanted '%s', got '%s'", expectedExecOutput, proxyExecOutput)
+				}
+
+				// Verify the proxy server logs saw the connection
+				expectedProxyLog := fmt.Sprintf("Sending request GET %s/api", apiserverAddr)
+				rawProxyLog := strings.Split(runKubectl("log", "goproxy", fmt.Sprintf("--namespace=%v", ns)), "\n")
+				sort.Sort(sort.Reverse(sort.StringSlice(rawProxyLog)))
+				proxyLog := rawProxyLog[2]
```

Comment:
I originally went that route. The problem was that the logs continue to build up meaning that if the first ```proxyVar``` is successful then ```strings.Contains``` will always say the following ones were as well. Another option would be to redeploy the ```goproxy``` pod for each ```proxyVar``` but I fear it would add significant time to the test.